src/modules/api/pipeline-task/routes/task.py

```diff
@@ -107,13 +92,22 @@ def update_task():
 
   # Wrap generic exceptions in an Internal Error class
   except Exception as ex:
-    raise APIInternalError('Error updating pipeline operation record', call_id) from ex
+    raise APIInternalError('Error getting pipeline runner', call_id) from ex
+
+  # Make sure a job execution is specified
+  if exec_id is None:
+    raise APIBadRequestError('Operation name must specify a job execution.', call_id)
+
+
+  # Get the current status of the job, updating the PipelineOperation record implicitly
+  logger.debug(f"[{ call_id }] Checking the operation status and updating the PipelineOperation record...")
+  status = runner.check_status(exec_id)
 
 
   # Update all linked report entities
   try:
-    logger.debug(f"[{ call_id }] Updating all linked status records for operation {op}: {dict(op)}")
-    update_all_linked_status_records(op['operation_kind'], op['operation'])
+    logger.debug(f"[{ call_id }] Updating all linked status records to status { status }...")
+    update_all_linked_status_records(runner, exec_id, status)
```

Comment:
The `update_all_linked_status_records` function has been rewritten to use the `Runner` directly, instead of the `PipelineOperation`.  In a future PR, this is a good candidate for a class method on `JobPipeline` -- looking up all `Report`s that match a given `Runner` and updating them.